py/trtorch/csrc/utils.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+#include <pybind11/numpy.h>
+#include <pybind11/pybind11.h>
+
+#include <functional>
+#include <iostream>
+#include <string>
+#include "NvInfer.h"
+
+namespace tensorrt {
+namespace utils {
+
+namespace py = pybind11;
+
+// Returns the size in bytes of the specified data type.
+inline size_t size(nvinfer1::DataType type) {
+  switch (type) {
+    case nvinfer1::DataType::kFLOAT:
+      return 4;
+    case nvinfer1::DataType::kHALF:
+      return 2;
+    case nvinfer1::DataType::kINT8:
+      return 1;
+    case nvinfer1::DataType::kINT32:
+      return 4;
+    case nvinfer1::DataType::kBOOL:
+      return 1;
+  }
+  return -1;
+}
+
+// Converts a TRT datatype to its corresponding numpy dtype.
+inline py::dtype nptype(nvinfer1::DataType type) {
+  switch (type) {
+    case nvinfer1::DataType::kFLOAT:
+      return py::dtype("f4");
+    case nvinfer1::DataType::kHALF:
+      return py::dtype("f2");
+    case nvinfer1::DataType::kINT8:
+      return py::dtype("i1");
+    case nvinfer1::DataType::kINT32:
+      return py::dtype("i4");
+    case nvinfer1::DataType::kBOOL:
+      return py::dtype("b1");
+  }
+  return py::dtype("unknown");
+}
+
+// Returns the TRT type corresponding to the specified numpy type.
+inline nvinfer1::DataType type(const py::dtype& type) {
+  if (type.is(py::dtype("f4"))) {
+    return nvinfer1::DataType::kFLOAT;
+  } else if (type.is(py::dtype("f2"))) {
+    return nvinfer1::DataType::kHALF;
+  } else if (type.is(py::dtype("i4"))) {
+    return nvinfer1::DataType::kINT32;
+  } else if (type.is(py::dtype("i1"))) {
+    return nvinfer1::DataType::kINT8;
+  } else if (type.is(py::dtype("b1"))) {
+    return nvinfer1::DataType::kBOOL;
+  }
+  std::cout << "[ERROR] Unsupported numpy data type: " << type.kind() << type.itemsize() * 8

```

Comment:
Use TRTorch errors here

Reply:
Fixed it. This was inherited from TRT source code. I cleaned it up  to be as minimal and relevant as possible.